funcx_endpoint/tests/test_rabbit_mq/test_endpoint.py

```diff
@@ -0,0 +1,124 @@
+import json
+import logging
+import multiprocessing
+import random
+import uuid
+
+import pika
+
+from funcx_endpoint.endpoint.rabbit_mq import (
+    ResultQueuePublisher,
+    ResultQueueSubscriber,
+    TaskQueuePublisher,
+    TaskQueueSubscriber,
+)
+
+LOG_FORMAT = "%(levelname) -10s %(asctime)s %(name) -20s %(lineno) -5d: %(message)s"
+logging.basicConfig(level=logging.INFO, format=LOG_FORMAT)
+logger = logging.getLogger(__name__)
+
+ENDPOINT_ID = "231fab4e-630a-4d76-bbbb-cf0b4aedbdf9"
+CONN_PARAMS = pika.URLParameters("amqp://guest:guest@localhost:5672/")
+
+
+def start_task_q_subscriber(
+    endpoint_id: str,
+    out_queue: multiprocessing.Queue,
+    disconnect_event: multiprocessing.Event,
+):
+
+    task_q = TaskQueueSubscriber(
+        CONN_PARAMS,
+        external_queue=out_queue,
+        kill_event=disconnect_event,
+        endpoint_uuid=endpoint_id,
+    )
+    task_q.start()
+    return task_q
```

Comment:
Each test module seems to have duplicates of this and several related functions. Plus there's consistent teardown for this which you want to run.

I'd wrap this as a fixture provided by the conftest. e.g.
```python
@pytest.fixture
def task_queue_subscriber():
    task_q = TaskQueueSubscriber(...)
    task_q.start()
    yield
    task_q.close()
```

Reply:
I edited the above slightly when I realized it needs to yield back the kill event too. It's probably not exactly what you want.
If the event should be supplied by the test, it can still be a function-fixture which sets a finalizer. e.g.
```python
@pytest.fixture
def task_queue_subscriber(request):
    def func(kill_event):
        kill_event = multiprocessing.Event()
        task_q = TaskQueueSubscriber(..., kill_event=kill_event)
        task_q.start()
        request.addfinalizer(task_q.close)
        return task_q

    return func
```